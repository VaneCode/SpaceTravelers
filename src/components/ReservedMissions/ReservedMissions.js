// Import
import Card from 'react-bootstrap/Card';
import ListGroup from 'react-bootstrap/ListGroup';
import { useSelector, shallowEqual } from 'react-redux';

const ReservedMissions = () => {
  // Get the list of rockets from the state
  const missionsStatus = useSelector((state) => state.missions, shallowEqual);
  // Get the reserved rockets
  const joined = missionsStatus.mission.filter((mission) => mission.join);
  return (
    <Card style={{ width: '50%', border: 'none' }}>
      <Card.Title>My missions</Card.Title>
      <Card style={{ width: '100%', marginTop: '0.5rem' }}>
        <ListGroup variant="flush">
          {joined.length === 0 ? (
            <ListGroup.Item>Not joined missions.</ListGroup.Item>
          ) : (
            joined.map((mission) => (
              <ListGroup.Item key={mission.mission_id}>
                {mission.mission_name}
              </ListGroup.Item>
            ))
          )}
        </ListGroup>
      </Card>
    </Card>
  );
};

export default ReservedMissions;
