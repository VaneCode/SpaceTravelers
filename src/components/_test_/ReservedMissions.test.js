import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import ReservedMissions from '../ReservedMissions/ReservedMissions';

describe('Reserved missions test', () => {
  test('snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <ReservedMissions />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
