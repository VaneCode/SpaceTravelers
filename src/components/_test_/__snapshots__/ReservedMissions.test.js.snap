// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reserved missions test snapshot 1`] = `
<div
  className="card"
  style={
    Object {
      "border": "none",
      "width": "50%",
    }
  }
>
  <div
    className="card-title h5"
  >
    My missions
  </div>
  <div
    className="card"
    style={
      Object {
        "marginTop": "0.5rem",
        "width": "100%",
      }
    }
  >
    <div
      className="list-group list-group-flush"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        Not joined missions.
      </div>
    </div>
  </div>
</div>
`;
